### PR TITLE
fix(): Fix the situation where undefined + char exists when calculating couple

### PR DESCRIPTION
**Pull request by ccOfHome**
## Description
`couple = previousChar + _char` will not cause any calculation problems, but it can be optimized to avoid misunderstandings during the development process
#10815

### CODE REVIEW

**Comment by Smrtnyk** [comment]
Is it possible to add any meaningful tests for this?
It clearly must be fixing some scenario

**Comment by ccOfHome** [comment]
Sure, but I'm not very familiar with this aspect.
Please check if the following code is okay

> e2e/tests/visual-output/rendering/testcases/text.ts

```
{
    title: 'Text with edge case: calculating coupleWidth, undefined + char',
    golden: 'textCalculatingCoupleWidth.png',
    percentage: 0.02,
    disabled: 'node',
    size: [200, 200],
    async renderFunction(canvas, fabric) {
      const t1 = new fabric.FabricText('AB AB\nBA BA', {
        left: 10,
        top: 10,
        fontFamily: 'Arial',
        fontSize: 64,
        objectCaching: false,
        strokeWidth: 0,
      });
      t1.positionByLeftTop(new fabric.Point(10, 10));

      const t2 = new fabric.FabricText('ABA', {
        left: 10,
        top: 110,
        fontFamily: 'Arial',
        fontSize: 64,
        objectCaching: false,
        strokeWidth: 0,
      });
      t2.positionByLeftTop(new fabric.Point(10, 110));

      canvas.add(t1, t2);
    },
  },
```

**Comment by asturur** [comment]
For testing look at the file `Text.spec.ts` sibling of the file you just fixed.

look for this test here

```ts
  describe('measuring, splitting', () => {
    it('measuring', () => {
      cache.clearFontCache();
      const zwc = '\u200b';
      const text = new FabricText('');
      const style = text.getCompleteStyleDeclaration(0, 0);
      const measurement = text._measureChar('a', style, zwc, style);
      expect(measurement).toMatchSnapshot(roundSnapshotOptions);
      expect(measurement).toEqual(text._measureChar('a', style, zwc, style));
    });

    it('splits into lines', () => {
      const text = new FabricText('test foo bar-baz\nqux');
      expect(text._splitTextIntoLines(text.text)).toMatchSnapshot();
    });
  });
```

double the measuring test, change its title to measuring a single char, and modify the test and expectations. ( remove the zwc for example )

you want to test 2 things:

_measureChar returns a different ( or not ) value after your change.

cache.charWidthsCache has only a key for the char you measured ( example 'a' ) and not also for `undefineda` couple of chars.

a unit test is sufficient.

To run the test write `npm run test:vitest Text.spec.ts` and you should be able to quickly run the unit tests for that file.

If you are not familiar with unit tests, this is a valuable thing to know.

File: src/shapes/Text/Text.spec.ts [hunk 1]
```diff
@@ -1320,4 +1320,21 @@ describe('FabricText', () => {
...
+      const text = new FabricText('');
+      const style = text.getCompleteStyleDeclaration(0, 0);
+      const measurement1 = text._measureChar('a', style, '', style);
+      const measurement2 = text._measureChar('a', style, '', style);
+      expect(measurement1).toEqual(measurement2);
+      const cacheKeys = cache.charWidthsCache
+        .get(text.fontFamily.toLowerCase())
+        ?.get('normal_normal')
+        ?.keys();
+      expect(cacheKeys?.next().value).not.toBe('undefineda');
```
Comment (Smrtnyk) [inline src/shapes/Text/Text.spec.ts:1336]: `undefineda` is this a typo? did you mean `toBeDefined()` instead?

Comment (ccOfHome) [reply src/shapes/Text/Text.spec.ts:1336]: `undefineda` might appear as a key, so I used strings for comparison. This might not be the best code; do you have any suggestions?

Comment (Smrtnyk) [reply src/shapes/Text/Text.spec.ts:1336]: ah sorry, I got confused, you are right
I didn't get the test at first

**Comment by asturur** [comment]
Thankyou @ccOfHome  and @Smrtnyk
